src/OmniSharp.DotNetTest/TestFrameworks/TestFramework.cs

```diff
@@ -7,36 +7,39 @@ namespace OmniSharp.DotNetTest.TestFrameworks
 {
     internal abstract class TestFramework
     {
-        private static readonly ImmutableDictionary<string, TestFramework> s_frameworks;
+        private static readonly ImmutableArray<TestFramework> s_frameworks;
```

Comment:
It's a neat idea to make this extensible, but with the amount of effort that would need to be put in to add another framework makes this a pointless optimization. It causes `GetFrameworks()` to allocate an iterator on every invocation (which is often), and even in the worst case is `GetFramework(string)` implemented as a foreach loop over an array will be twice as fast as a dictionary lookup: https://gist.github.com/333fred/2c259fff1ca0e074adead95f01db70ba.